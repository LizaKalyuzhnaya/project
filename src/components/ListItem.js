import React from 'react'
import PropTypes from 'prop-types'


class ListItem extends React.Component {
  render() {
    const { provided, innerRef } = this.props;
    return (
      <li {...provided.draggableProps}
          {...provided.dragHandleProps}
          ref={innerRef} 
          className = 'app__search__list-place__item' >
        <p className = 'app__search__list-place__item__text' >{this.props.text}</p>
        <div onClick = {this.props.onClick}>
          <svg className = 'app__search__list-place__item__svg' width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path fillRule="evenodd" clipRule="evenodd" d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z" fill="#007AFF"/>
          </svg>
        </div>
      </li>
    )
  }
}

ListItem.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default ListItem