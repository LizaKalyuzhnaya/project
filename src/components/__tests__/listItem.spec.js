import React from 'react'
import { shallow } from 'enzyme'
import ListItem from '../ListItem'

describe('List items', () => {
    const props = {
        text: 'a',
        provided: {}
    }
    const myComponent = shallow(<ListItem {...props} />)

    it('render initial', () => {
            expect(myComponent.find('p').text()).toEqual(props.text);
            expect(myComponent.find('svg')).toHaveLength(1);
    })

    it('renders properly', () => {
        expect(myComponent).toMatchSnapshot()
    })
})
