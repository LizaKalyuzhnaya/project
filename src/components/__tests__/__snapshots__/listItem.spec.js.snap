// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List items renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    provided={Object {}}
    text="a"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="app__search__list-place__item__text"
        >
          a
        </p>,
        <div>
          <svg
            className="app__search__list-place__item__svg"
            fill="none"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z"
              fill="#007AFF"
              fillRule="evenodd"
            />
          </svg>
        </div>,
      ],
      "className": "app__search__list-place__item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "a",
          "className": "app__search__list-place__item__text",
        },
        "ref": null,
        "rendered": "a",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="app__search__list-place__item__svg"
            fill="none"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z"
              fill="#007AFF"
              fillRule="evenodd"
            />
          </svg>,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <path
              clipRule="evenodd"
              d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z"
              fill="#007AFF"
              fillRule="evenodd"
            />,
            "className": "app__search__list-place__item__svg",
            "fill": "none",
            "height": "14",
            "viewBox": "0 0 14 14",
            "width": "14",
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z",
              "fill": "#007AFF",
              "fillRule": "evenodd",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          "type": "svg",
        },
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="app__search__list-place__item__text"
          >
            a
          </p>,
          <div>
            <svg
              className="app__search__list-place__item__svg"
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z"
                fill="#007AFF"
                fillRule="evenodd"
              />
            </svg>
          </div>,
        ],
        "className": "app__search__list-place__item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "a",
            "className": "app__search__list-place__item__text",
          },
          "ref": null,
          "rendered": "a",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="app__search__list-place__item__svg"
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z"
                fill="#007AFF"
                fillRule="evenodd"
              />
            </svg>,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <path
                clipRule="evenodd"
                d="M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z"
                fill="#007AFF"
                fillRule="evenodd"
              />,
              "className": "app__search__list-place__item__svg",
              "fill": "none",
              "height": "14",
              "viewBox": "0 0 14 14",
              "width": "14",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "clipRule": "evenodd",
                "d": "M1.09325 0.0324857L0.0325866 1.09315L5.68944 6.75L0.0325866 12.4069L1.09325 13.4675L6.7501 7.81066L12.407 13.4675L13.4676 12.4069L7.81076 6.75L13.4676 1.09315L12.407 0.0324857L6.7501 5.68934L1.09325 0.0324857Z",
                "fill": "#007AFF",
                "fillRule": "evenodd",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            "type": "svg",
          },
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
